bibliopixel/drivers/spi_driver_base.py

```diff
@@ -1,75 +1,155 @@
-import os, sys, time
+import fcntl
```

Comment:
This is mildly pedantic... but this import is going to cause a hard exception on Windows. Hey... come back! Bare with me... I know, SPI drivers aren't valid on Windows and not even on all Linux. My concern here is for a *graceful* failure with pertinent details on unsupported operating systems. At the very least I would check if the system is 'nt' or 'darwin' upon import of this file (at the top) and immediately fail with a message stating that SPI drivers are unsupported on these systems. The issue here is the fact that ```fcntl``` does not exist on Windows (prob does on Mac, not sure) so this will just hard fail. You won't even be able to import it to run the tests with the dummy SPI mode. Yes, I don't actually test or develop on Windows but I just see it has bad form to not have every single module of this library importable on every supported OS without at least throwing a clean and clear error message.